kasa/device.py

```diff
@@ -318,10 +318,10 @@ def features(self) -> dict[str, Feature]:
 
     def _add_feature(self, feature: Feature):
         """Add a new feature to the device."""
-        desc_name = feature.name.lower().replace(" ", "_")
-        if desc_name in self._features:
-            raise KasaException("Duplicate feature name %s" % desc_name)
-        self._features[desc_name] = feature
+        if feature.id in self._features:
```

Comment:
I moved the id generation into `Feature` to allow overriding it, which we may need to do to keep backwards compat with homeassistant's emeter sensors.